### DISCUSSION TITLE
Getting rid of xochitl dependency

### DISCUSSION THREAD

**Discussion by floers** [timower/rM2-stuff#41] [General]
Hey, I was wondering whether it might be possible to get rid of the remaining xochitl dependencies. What exactly does this generator thread do? How were you able to get so far at all? I can't find much information about this whole stuff. I would like to help if I can.

**Comment by timower** [comment]
Those threads are the routines that actually update the screen. The generator thread takes the framebuffer images and transforms it before blitting it on the memory mapped hardware framebuffer. The 'vsync' thread is responsible for switching between the different generated images. 
These transformations are not simple, they depend on the current state of the display, the temperature, the chosen waveform and update mode, ...
So re-implementing them is a lot of work, @matteodelabre started a project at https://github.com/matteodelabre/waved to do this.

The problem is that some xochitl updates contained improvements to these functions, so using an older version for their threads, with a newer xochitl app would break. So we have to keep in sync, and keep updating the hooks.

**Reply by floers** [reply]
Hm I see... waved seems to be not up to date. At least I can't run it on my rm2. I will dig in to the code a bit. Maybe I can help. Thanks for answering my question :).
